pkg/trtl/peers.go

```diff
@@ -53,12 +40,47 @@ func (p *PeerService) GetPeers(ctx context.Context, in *peers.PeersFilter) (out
 	return out, nil
 }
 
-// AddPeers adds a peer and returns a report of the status of all peers in the network
+// AddPeers adds a peer and returns a report of the status of all peers in the network.
 func (p *PeerService) AddPeers(ctx context.Context, in *peers.Peer) (out *peers.PeersStatus, err error) {
-	// CreatePeer handles possibility of an already-existing or previously deleted peer
-	if _, err := p.store.CreatePeer(in); err != nil {
-		log.Error().Err(err).Msg("unable to add peer")
-		return nil, status.Error(codes.InvalidArgument, "invalid peer; could not be added")
+	// Compute the key from the Peer method to ensure consistent key access.
+	// If the key is empty, that means the peer ID == 0; return an error.
+	key := in.Key()
+	if key == "" {
+		return nil, status.Error(codes.InvalidArgument, "a valid peer ID is required")
+	}
```

Comment:
Because the Key is based on the PID, it is impossible to create a Peer with a duplicate PID.